packages/react/src/wishlists/hooks/types/useWishlistSets.ts

```diff
@@ -0,0 +1,18 @@
+import type { SetsState } from '@farfetch/blackout-redux/wishlists/types';
+import type { WishlistSets } from '@farfetch/blackout-client/wishlists/types';
+import type {
+  WishlistSetsErrors,
+  WishlistSetsHydrated,
+} from '@farfetch/blackout-redux/wishlists/selectors/types';
+
+export type UseWishlistSets = () => {
+  allWishlistSetsErrors: WishlistSetsErrors | undefined;
+  areLoading: SetsState['isLoading'];
+  error: SetsState['error'] | undefined;
+  fetchWishlistSets: () => Promise<WishlistSets>;
+  isAnyWishlistSetLoading: boolean;
+  isAnyWishlistSetWithError: boolean;
+  resetWishlistSets: () => void;
+  resetWishlistSetsState: (fieldsToReset?: string[]) => void;
+  wishlistSets: WishlistSetsHydrated | undefined;
```

Comment:
If you change the `WishlistSetsHydrated`, you won't need the `| undefined` here 😉

Reply:
Done 🙂